### PR TITLE
treewide: remove some libglade dependents

### PR DESCRIPTION
**Pull request by piegamesde**
###### Motivation for this change

- #148779 Python 2 has reached its end of life and packages depending on it shall be patched or removed
- `libglade` depends on Python 2. The functionality of libglade itself has been subsumed as "Builder" in Gtk3. Gtk2 is end of life as well, btw.
- Dropped a some packages that depend on `libglade`, and their transitive dependencies.
- Notable drops:
  - `libgtkmm`, the C++ bindings, and all applications using it
  - `libgksu` and `gksu`
  - gtk-sharp and thus all Gtk2 applications written in C#
  - `lablgtk` and all Gtk2 application written in OCaml
  - Various random applications ("leaf packages")

Some packages are fixed instead of being removed. Some more can be, I'll replace the drop commits with fix commits as we go. Help welcome (see #155173 for example).

This PR is still a WIP, but it has mostly reached its final form. Commits that are incomplete are marked with TODO in the message.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -661,6 +669,7 @@ mapAliases ({
   nylas-mail-bin = throw "nylas-mail-bin was deprecated on 2019-09-11: abandoned by upstream";
   oauth2_proxy = oauth2-proxy; # added 2021-04-18
   opencascade_oce = opencascade; # added 2018-04-25
+  obconf = throw "obconf has been removed"; # added 2021-01-16
```
Comment (Mindavi) [inline pkgs/top-level/aliases.nix:672]: Make that 2022 :)

Comment (piegamesde) [reply pkgs/top-level/aliases.nix:672]: Oh no, muscle memory :see_no_evil: And I consistently did it wrong everywhere

**Comment by piegamesde** [comment]
Pinging maintainers of the directly affected packges:

- `fslint` @dasj19
- `pygtk` (unmaintained)
- `Gtk2GladeXML` (unmaintained)
- `yarssr`	@xurei
- `prooftree`	@jwiegley
- `ocamlgraph_gtk`	@kkallio
- `camlimages`	@vbgl, @mt-caret
- `gtk-sharp` (unmaintained)
- `ocaml_cairo` (unmaintained)
- `monotoneVis`	@raskin
- `satysfi`	@mt-caret @marsam
- `glsurf` (unmaintained)
- `tilp2`	@luc65r
- `gfm`	@siraben, @luc65r
- `libglademm` (unmaintained)
- `fme`	@AndersonTorres
- `timekeeper`	@bburdette
- `guile-gnome` @vyp
- `libgksu` @romildo
- `gksu` @romildo
- `obconf` @nathanielbaxter
- `monodevelop` @obadz
- `tomboy` @stesie
- `pdfmod` @obadz
- `gnome-sharp` @vbgl
- Generally: @NixOS/gnome

**Comment by mweinelt** [comment]
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>supertux-editor</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>picoscope</li>
    <li>snapdragon-profiler</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>appindicator-sharp</li>
    <li>bless</li>
    <li>dotnetPackages.MonoAddins</li>
    <li>gio-sharp</li>
    <li>gtk-sharp-2_0</li>
    <li>gtk-sharp-3_0</li>
    <li>gtk-sharp-beans</li>
    <li>hyena</li>
    <li>mono-addins</li>
    <li>notify-sharp</li>
    <li>sparkleshare</li>
    <li>webkit2-sharp</li>
  </ul>
</details>

**Comment by piegamesde** [comment]
As far as I can tell are both `picoscope` and `snapdragon-profiler` are proprietary and fail to build because they are missing the sources.

**Comment by bburdette** [comment]
thx for the ping.  you're welcome to remove timekeeper if you like; I've written a new timeclock since then and I don't have plans to upgrade this old one.

**Comment by vbgl** [comment]
Please double-check your list. I think that a few packages have been included by mistake and should be kept in `nixpkgs` (to be understrood as: “not removed as part of this PR dedicated to `libglade`”): prooftree, ocamlgraph_gtk, camlimages, ocaml_cairo, monotoneViz, satysfi, glsurf.

**Review by dasj19 — COMMENTED** [review]
hi, I checked `fslint` and upstream does not support `python3` so I guess it is safe to remove the package.
Thanks for taking care of this, and thanks for adding an alias.

**Comment by piegamesde** [comment]
@vbgl thanks for having a look, those packages were deleted in a previous iteration of the PR and I forgot to update that part.

----


I am confused about `gtktop` though: its "drop" commit only contained the alias somehow, and I cannot find that file anymore at all. I assume that it was already dropped in an independent PR?

**Comment by vbgl** [comment]
You can run something like `git log --oneline | grep gtktop ` to quickly discover commit 1c1b91861a5 (and PR #155204 it belongs to).

**Comment by piegamesde** [comment]
@NixOS/darwin-maintainers Eval appears to fail on `aarch64-darwin`. At first I thought it was a flaky CI, but same thing on the second try. No specific error message from what I can tell. Any ideas?

**Comment by reckenrode** [comment]
> @NixOS/darwin-maintainers Eval appears to fail on `aarch64-darwin`. At first I thought it was a flaky CI, but same thing on the second try. No specific error message from what I can tell. Any ideas?

I get the following failure. There appears to be a reference to `gfm` in `top-level/all-packages.nix`.

    error: getting status of '/Users/reckenrode/.cache/nixpkgs-review/pr-155061/nixpkgs/pkgs/applications/science/math/gfm': No such file or directory

       … while evaluating 'makeOverridable'

       at /Users/reckenrode/.cache/nixpkgs-review/pr-155061/nixpkgs/lib/customisation.nix:67:24:

           66|   */
           67|   makeOverridable = f: origArgs:
             |                        ^
           68|     let

       … from call site

       at /Users/reckenrode/.cache/nixpkgs-review/pr-155061/nixpkgs/lib/customisation.nix:121:8:

          120|       auto = builtins.intersectAttrs (lib.functionArgs f) autoArgs;
          121|     in makeOverridable f (auto // args);
             |        ^
          122|

       … while evaluating 'callPackageWith'

       at /Users/reckenrode/.cache/nixpkgs-review/pr-155061/nixpkgs/lib/customisation.nix:117:35:

          116|   */
          117|   callPackageWith = autoArgs: fn: args:
             |                                   ^
          118|     let

       … from call site

       at /Users/reckenrode/.cache/nixpkgs-review/pr-155061/nixpkgs/pkgs/top-level/all-packages.nix:16860:9:

        16859|
        16860|   gfm = callPackage ../applications/science/math/gfm { };
             |         ^
        16861|
    https://github.com/NixOS/nixpkgs/pull/155061 failed to build

**Review by dasj19 — COMMENTED** [review]
Tomboy was removed in this PR: https://github.com/NixOS/nixpkgs/pull/156979

**Comment by dasj19** [comment]
I could build this by removing the following files from https://github.com/NixOS/nixpkgs/pull/155061.patch

```
pkgs/applications/misc/tomboy/default.nix
pkgs/applications/editors/monodevelop/git-revert-12d610fb3f6dce121df538e36f21d8c2eeb0a6e3.patch
```
There is some issue in applying the patch file on the latest master branch. The two files I pointed are the reason for the issues.

**Comment by piegamesde** [comment]
@dasj19 I rebased on master and only got conflicts for the `tomboy` commit. I don't know where your problems with the `monodevelop` file come from.

**Review by dasj19 — APPROVED** [review]
Now it builds. I guess the error with the monodevelop file was from the way I was applying the patch.

But nevermind, everything works now

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 155061` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>supertux-editor</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>picoscope</li>
    <li>snapdragon-profiler</li>
  </ul>
</details>
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>appindicator-sharp</li>
    <li>bless</li>
    <li>dotnetPackages.MonoAddins</li>
    <li>gio-sharp</li>
    <li>gtk-sharp-2_0</li>
    <li>gtk-sharp-3_0</li>
    <li>gtk-sharp-beans</li>
    <li>hyena</li>
    <li>mono-addins</li>
    <li>notify-sharp</li>
    <li>sparkleshare</li>
    <li>webkit2-sharp</li>
  </ul>
</details>

**Pull request merged**